public/app/core/components/AppChrome/ExtensionSidebar/ExtensionToolbarItem.tsx

```diff
@@ -0,0 +1,59 @@
+import { useState } from 'react';
+
+import { Dropdown, Menu, ToolbarButton } from '@grafana/ui';
+
+import { getComponentIdFromComponentMeta, useExtensionSidebarContext } from './ExtensionSidebarProvider';
+
+export function ExtensionToolbarItem() {
+  const [isOpen, setIsOpen] = useState(false);
+  const { availableComponents, dockedComponentId, setDockedComponentId } = useExtensionSidebarContext();
+
+  if (availableComponents.size === 0) {
+    return null;
+  }
+
+  // get a flat list of all components with their pluginId
+  const components = Array.from(availableComponents.entries()).flatMap(([pluginId, { exposedComponents }]) =>
+    exposedComponents.map((c) => ({ ...c, pluginId }))
+  );
+
+  if (components.length === 0) {
+    return null;
+  }
+
+  if (components.length === 1) {
+    return (
+      <ToolbarButton
+        icon="web-section"
+        variant={dockedComponentId ? 'active' : 'default'}
+        tooltip={components[0].description}
+        onClick={() => {
```

Comment:
I _think_ there's a bit of a mismatch going on between local state here and the context state which is causing the sidebar panel to remain open when clicking this button multiple times. Perhaps using isOpen from the context and checking that here might make this toggle-able again? Something like:

```
if (isOpen) {
  setDockedComponentId(undefined);
} else {
  setDockedComponentId(getComponentIdFromComponentMeta(components[0].pluginId, components[0]));
}
```

Reply:
Nice, thank you - that makes a lot of sense. Fixed in [aa8e948](https://github.com/grafana/grafana/pull/102626/commits/aa8e948fdbd54c39ebe720cba012b231ee121bfc)